src/transforms/aws_ec2_metadata.rs

```
@@ -281,66 +294,37 @@ impl MetadataClient {
-        parts.path_and_query = Some(DYNAMIC_DOCUMENT.clone());
-        let uri = Uri::from_parts(parts)?;
-
-        let req = Request::get(uri)
-            .header(TOKEN_HEADER.as_ref(), token.as_ref())
-            .body(Body::empty())?;
-
-        let res = self.client.send(req).await?;
-
-        if res.status() != StatusCode::OK {
-            warn!(message="Identity document request failed.", status = %res.status());
-            return Ok(None);
-        }
-
-        let body = body_to_bytes(res.into_body()).await?;
-
-        serde_json::from_slice(&body[..])
-            .map_err(Into::into)
-            .map(Some)
+    pub async fn get_document(&mut self) -> Option<IdentityDocument> {
+        self.get_document_error()
+            .await
+            .with_context(|| FetchIdentityDocument {})
+            .map_err(|err| {
+                emit!(AwsEc2MetadataRequestFailed {
+                    path: DYNAMIC_DOCUMENT.as_str(),
+                    error: err.into(),
+                });
+            })
+            .ok()
     }
 
-    pub async fn get_metadata(
-        &mut self,
-        path: &PathAndQuery,
-    ) -> Result<Option<Bytes>, crate::Error> {
-        let token = self.get_token().await?;
-
-        let mut parts = self.host.clone().into_parts();
-
-        parts.path_and_query = Some(path.clone());
-
-        let uri = Uri::from_parts(parts)?;
-
-        debug!(message = "Sending metadata request.", %uri);
-
-        let req = Request::get(uri)
-            .header(TOKEN_HEADER.as_ref(), token.as_ref())
-            .body(Body::empty())?;
-
-        let res = self.client.send(req).await?;
-
-        if StatusCode::OK != res.status() {
-            warn!(message="Metadata request failed.", status = %res.status());
-            return Ok(None);
-        }
-
-        let body = body_to_bytes(res.into_body()).await?;
-
-        Ok(Some(body))
+    pub async fn get_metadata(&mut self, path: &PathAndQuery) -> Option<Bytes> {
```

Similarly, swallowing the error prevents action on it.